marlowe-playground-client/src/MainFrame/Types.purs

```diff
@@ -42,13 +42,23 @@ data ModalView
   | OpenDemo
   | RenameProject
   | SaveProjectAs
-  | GithubLogin
+  | GithubLogin Action
 
 derive instance genericModalView :: Generic ModalView _
 
 instance showModalView :: Show ModalView where
-  show = genericShow
-
+  show NewProject = "NewProject"
+  show OpenProject = "OpenProject"
+  show OpenDemo = "OpenDemo"
+  show RenameProject = "RenameProject"
+  show SaveProjectAs = "SaveProjectAs"
+  show (GithubLogin _) = "GithubLogin"
+
+-- Before adding the intended action to GithubLogin, this instance was being
+-- handled by the genericShow. Action does not have a show instance so genericShow
+-- does not work. For the moment I've made a manual instance, but not sure why
+-- ModalView requires show, or if we should make Action an instance of Show
+-- show = genericShow
```

Comment:
we use this in analytics events. My preference would be to make a `Show Action`

Reply:
Trying to do this it fails to find a Show instance for Halogen's SubscriptionId. For this PR I'll leave it as it is, later on we can see how to approach this.

Is it possible to mix genericShow with some exceptions?

Reply:
yeah I guess you can pattern match the definition of `show` and only use `genericShow` on the ones that work.

Reply:
@shmish111 not sure if this was what you meant, but it gives me the same error :/ 
![Screen Shot 2020-11-24 at 19 11 46](https://user-images.githubusercontent.com/2634059/100157251-fbcedc00-2e88-11eb-9c59-1dae29d61c1a.png)

Reply:
yeah that's what I meant. well that's a shame 👎